src/components/dms/MessageProfileButton.tsx

```diff
@@ -59,14 +51,15 @@ export function MessageProfileButton({
       logEvent('chat:create', {logContext: 'ProfileHeader'})
       initiateConvo([profile.did])
     }
-  }, [
-    needsEmailVerification,
-    verifyEmailControl,
-    navigation,
-    profile.did,
-    initiateConvo,
-    convoAvailability,
-  ])
+  }, [navigation, profile.did, initiateConvo, convoAvailability])
+
+  const wrappedOnPress = requireEmailVerification(onPress, {
+    instructions: [
+      <Trans key="message">
+        Before you may message another user, you must first verify your email.
```

Comment:
```suggestion
        Before you can message another user, you must first verify your email.
```
nit: Slight suggested wording tweak that sounds a bit less formal imo